ICE: ast: `{{root}} != {{root}}`

<!--
[31mICE[0m: Rustc ./code.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_ast/src/ast.rs:198:9: 'assertion `left != right` failed'', 'thread 'rustc' ($TID) panicked at compiler/rustc_ast/src/ast.rs:198:9: 'assertion `left != right` failed''
File: /tmp/im/code.rs
-->

original:
````rust
mod A {
  use Iuse::
    { ::Fish};
  
}
mod A {
  mod Fish {
    use A::Fish;
  }
}

````

Version information
````
rustc 1.94.0-nightly (f2c70877a 2025-12-17)
binary: rustc
commit-hash: f2c70877a7dbc51cd98e2d5d25209b24d4a586d5
commit-date: 2025-12-17
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8

````



Command:
`/home/matthias/.rustup/toolchains/alt-master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0428]: the name `A` is defined multiple times
 --> code.rs:6:1
  |
1 | mod A {
  | ----- previous definition of the module `A` here
...
6 | mod A {
  | ^^^^^ `A` redefined here
  |
  = note: `A` must be defined only once in the type namespace of this module

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `Iuse`
 --> code.rs:2:7
  |
2 |   use Iuse::
  |       ^^^^ use of unresolved module or unlinked crate `Iuse`
  |
help: you might be missing a crate named `Iuse`, add it to your project and import it in your code
  |
1 + extern crate Iuse;
  |


thread 'rustc' (561535) panicked at compiler/rustc_ast/src/ast.rs:198:9:
assertion `left != right` failed
  left: "{{root}}"
 right: "{{root}}"
stack backtrace:
   0:     0x7f2c515211e0 - std[497fd58be37922e3]::backtrace_rs::backtrace::libunwind::trace
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
   1:     0x7f2c515211e0 - std[497fd58be37922e3]::backtrace_rs::backtrace::trace_unsynchronized::<std[497fd58be37922e3]::sys::backtrace::_print_fmt::{closure#1}>
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
   2:     0x7f2c515211e0 - std[497fd58be37922e3]::sys::backtrace::_print_fmt
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/sys/backtrace.rs:74:9
   3:     0x7f2c515211e0 - <<std[497fd58be37922e3]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[90b112ef4e25e834]::fmt::Display>::fmt
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/sys/backtrace.rs:44:26
   4:     0x7f2c4cbbb24e - <core[90b112ef4e25e834]::fmt::rt::Argument>::fmt
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/fmt/rt.rs:152:76
   5:     0x7f2c4cbbb24e - core[90b112ef4e25e834]::fmt::write
   6:     0x7f2c51539d5c - std[497fd58be37922e3]::io::default_write_fmt::<std[497fd58be37922e3]::sys::stdio::unix::Stderr>
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/io/mod.rs:614:11
   7:     0x7f2c51539d5c - <std[497fd58be37922e3]::sys::stdio::unix::Stderr as std[497fd58be37922e3]::io::Write>::write_fmt
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/io/mod.rs:1969:13
   8:     0x7f2c514e73b6 - <std[497fd58be37922e3]::sys::backtrace::BacktraceLock>::print
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/sys/backtrace.rs:47:9
   9:     0x7f2c514e73b6 - std[497fd58be37922e3]::panicking::default_hook::{closure#0}
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:292:27
  10:     0x7f2c51510801 - std[497fd58be37922e3]::panicking::default_hook
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:319:9
  11:     0x7f2c4df6ccc8 - <alloc[54afb4ad215f9a5d]::boxed::Box<dyn for<'a, 'b> core[90b112ef4e25e834]::ops::function::Fn<(&'a std[497fd58be37922e3]::panic::PanicHookInfo<'b>,), Output = ()> + core[90b112ef4e25e834]::marker::Sync + core[90b112ef4e25e834]::marker::Send> as core[90b112ef4e25e834]::ops::function::Fn<(&std[497fd58be37922e3]::panic::PanicHookInfo,)>>::call
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/alloc/src/boxed.rs:2220:9
  12:     0x7f2c4df6ccc8 - rustc_driver_impl[2248f57d38587f69]::install_ice_hook::{closure#1}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_driver_impl/src/lib.rs:1436:17
  13:     0x7f2c4df6ccc8 - <alloc[54afb4ad215f9a5d]::boxed::Box<rustc_driver_impl[2248f57d38587f69]::install_ice_hook::{closure#1}> as core[90b112ef4e25e834]::ops::function::Fn<(&dyn for<'a, 'b> core[90b112ef4e25e834]::ops::function::Fn<(&'a std[497fd58be37922e3]::panic::PanicHookInfo<'b>,), Output = ()> + core[90b112ef4e25e834]::marker::Sync + core[90b112ef4e25e834]::marker::Send, &std[497fd58be37922e3]::panic::PanicHookInfo)>>::call
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/alloc/src/boxed.rs:2220:9
  14:     0x7f2c4df6ccc8 - std[497fd58be37922e3]::panicking::update_hook::<alloc[54afb4ad215f9a5d]::boxed::Box<rustc_driver_impl[2248f57d38587f69]::install_ice_hook::{closure#1}>>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:235:47
  15:     0x7f2c51510b91 - <alloc[54afb4ad215f9a5d]::boxed::Box<dyn for<'a, 'b> core[90b112ef4e25e834]::ops::function::Fn<(&'a std[497fd58be37922e3]::panic::PanicHookInfo<'b>,), Output = ()> + core[90b112ef4e25e834]::marker::Sync + core[90b112ef4e25e834]::marker::Send> as core[90b112ef4e25e834]::ops::function::Fn<(&std[497fd58be37922e3]::panic::PanicHookInfo,)>>::call
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/alloc/src/boxed.rs:2220:9
  16:     0x7f2c51510b91 - std[497fd58be37922e3]::panicking::panic_with_hook
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:833:13
  17:     0x7f2c514e747a - std[497fd58be37922e3]::panicking::panic_handler::{closure#0}
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:698:13
  18:     0x7f2c514d2fe9 - std[497fd58be37922e3]::sys::backtrace::__rust_end_short_backtrace::<std[497fd58be37922e3]::panicking::panic_handler::{closure#0}, !>
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/sys/backtrace.rs:182:18
  19:     0x7f2c514e96bd - __rustc[d8d64968b6079d54]::rust_begin_unwind
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:689:5
  20:     0x7f2c4cbbbc0c - core[90b112ef4e25e834]::panicking::panic_fmt
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/panicking.rs:80:14
  21:     0x7f2c4cbbba93 - core[90b112ef4e25e834]::panicking::assert_failed_inner
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/panicking.rs:439:17
  22:     0x7f2c50a5ead9 - core[90b112ef4e25e834]::panicking::assert_failed::<rustc_span[7053f045cc1d2195]::symbol::Symbol, rustc_span[7053f045cc1d2195]::symbol::Symbol>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/panicking.rs:394:5
  23:     0x7f2c50890cf6 - rustc_ast[51cce773cc7ac4c5]::ast::join_path_idents::<rustc_span[7053f045cc1d2195]::symbol::Ident, alloc[54afb4ad215f9a5d]::vec::Vec<rustc_span[7053f045cc1d2195]::symbol::Ident>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_ast/src/ast.rs:198:9
  24:     0x7f2c50890cf6 - <rustc_resolve[a1515088aa319747]::Resolver>::report_privacy_error
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_resolve/src/diagnostics.rs:2275:24
  25:     0x7f2c507decf1 - <rustc_resolve[a1515088aa319747]::Resolver>::report_errors
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_resolve/src/diagnostics.rs:169:22
  26:     0x7f2c507decf1 - <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate::{closure#0}::{closure#7}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_resolve/src/lib.rs:1899:65
  27:     0x7f2c507decf1 - <rustc_data_structures[7f1c5b0d892434a8]::profiling::VerboseTimingGuard>::run::<(), <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate::{closure#0}::{closure#7}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_data_structures/src/profiling.rs:844:9
  28:     0x7f2c507decf1 - <rustc_session[5163d6f1fc0ec2cd]::session::Session>::time::<(), <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate::{closure#0}::{closure#7}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_session/src/utils.rs:17:50
  29:     0x7f2c507decf1 - <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_resolve/src/lib.rs:1899:27
  30:     0x7f2c508b5f5f - <rustc_data_structures[7f1c5b0d892434a8]::profiling::VerboseTimingGuard>::run::<(), <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate::{closure#0}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_data_structures/src/profiling.rs:844:9
  31:     0x7f2c508b5f5f - <rustc_session[5163d6f1fc0ec2cd]::session::Session>::time::<(), <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate::{closure#0}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_session/src/utils.rs:17:50
  32:     0x7f2c508b5f5f - <rustc_resolve[a1515088aa319747]::Resolver>::resolve_crate
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_resolve/src/lib.rs:1887:23
  33:     0x7f2c4eedcf0e - rustc_interface[240f805485c85136]::passes::configure_and_expand
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/passes.rs:304:14
  34:     0x7f2c4eede73a - rustc_interface[240f805485c85136]::passes::resolver_for_lowering_raw
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/passes.rs:799:17
  35:     0x7f2c5025e4d7 - rustc_query_impl[6d7566683bdc91cf]::query_impl::resolver_for_lowering_raw::dynamic_query::{closure#2}::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/plumbing.rs:298:9
  36:     0x7f2c5025e4d7 - rustc_query_impl[6d7566683bdc91cf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[6d7566683bdc91cf]::query_impl::resolver_for_lowering_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/plumbing.rs:582:18
  37:     0x7f2c506b9cdb - rustc_query_impl[6d7566683bdc91cf]::query_impl::resolver_for_lowering_raw::dynamic_query::{closure#2}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/plumbing.rs:659:25
  38:     0x7f2c506b9cdb - <rustc_query_impl[6d7566683bdc91cf]::query_impl::resolver_for_lowering_raw::dynamic_query::{closure#2} as core[90b112ef4e25e834]::ops::function::FnOnce<(rustc_middle[c1a3e287041cf856]::ty::context::TyCtxt, ())>>::call_once
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/ops/function.rs:250:5
  39:     0x7f2c502cb3c4 - <rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false> as rustc_query_system[86d8dbe0247c04e7]::query::config::QueryConfig<rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>>::compute
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/lib.rs:119:9
  40:     0x7f2c502cb3c4 - rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr::<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_system/src/query/plumbing.rs:506:72
  41:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context::<rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  42:     0x7f2c502cb3c4 - <std[497fd58be37922e3]::thread::local::LocalKey<core[90b112ef4e25e834]::cell::Cell<*const ()>>>::try_with::<rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context<rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/thread/local.rs:513:12
  43:     0x7f2c502cb3c4 - <std[497fd58be37922e3]::thread::local::LocalKey<core[90b112ef4e25e834]::cell::Cell<*const ()>>>::with::<rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context<rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/thread/local.rs:477:20
  44:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context::<rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  45:     0x7f2c502cb3c4 - <rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query::<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/plumbing.rs:169:13
  46:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::with_related_context::<<rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:112:9
  47:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::with_context::<rustc_middle[c1a3e287041cf856]::ty::context::tls::with_related_context<<rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:90:36
  48:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::with_context_opt::<rustc_middle[c1a3e287041cf856]::ty::context::tls::with_context<rustc_middle[c1a3e287041cf856]::ty::context::tls::with_related_context<<rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:79:18
  49:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::with_context::<rustc_middle[c1a3e287041cf856]::ty::context::tls::with_related_context<<rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:90:5
  50:     0x7f2c502cb3c4 - rustc_middle[c1a3e287041cf856]::ty::context::tls::with_related_context::<<rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:103:5
  51:     0x7f2c502cb3c4 - <rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt as rustc_query_system[86d8dbe0247c04e7]::query::QueryContext>::start_query::<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/plumbing.rs:155:9
  52:     0x7f2c502cb3c4 - rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job_non_incr::<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_system/src/query/plumbing.rs:506:22
  53:     0x7f2c502cb3c4 - rustc_query_system[86d8dbe0247c04e7]::query::plumbing::execute_job::<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt, false>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_system/src/query/plumbing.rs:442:9
  54:     0x7f2c502cb3c4 - rustc_query_system[86d8dbe0247c04e7]::query::plumbing::try_execute_query::<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt, false>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_system/src/query/plumbing.rs:385:13
  55:     0x7f2c504cc336 - rustc_query_system[86d8dbe0247c04e7]::query::plumbing::get_query_non_incr::<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_system/src/query/plumbing.rs:827:32
  56:     0x7f2c504cc336 - stacker[c1bc7817e21f8102]::maybe_grow::<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::get_query_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  57:     0x7f2c504cc336 - rustc_data_structures[7f1c5b0d892434a8]::stack::ensure_sufficient_stack::<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[86d8dbe0247c04e7]::query::plumbing::get_query_non_incr<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_data_structures/src/stack.rs:21:5
  58:     0x7f2c504cc336 - rustc_query_system[86d8dbe0247c04e7]::query::plumbing::get_query_non_incr::<rustc_query_impl[6d7566683bdc91cf]::DynamicConfig<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[6d7566683bdc91cf]::plumbing::QueryCtxt>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_system/src/query/plumbing.rs:827:5
  59:     0x7f2c504cc336 - rustc_query_impl[6d7566683bdc91cf]::query_impl::resolver_for_lowering_raw::get_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_query_impl/src/plumbing.rs:635:26
  60:     0x7f2c4f540c55 - rustc_middle[c1a3e287041cf856]::query::inner::query_get_at::<rustc_query_system[86d8dbe0247c04e7]::query::caches::SingleCache<rustc_middle[c1a3e287041cf856]::query::erase::Erased<[u8; 16usize]>>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/query/inner.rs:33:17
  61:     0x7f2c4f540c55 - <rustc_middle[c1a3e287041cf856]::query::plumbing::TyCtxtAt>::resolver_for_lowering_raw
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/query/plumbing.rs:415:31
  62:     0x7f2c4f540c55 - <rustc_middle[c1a3e287041cf856]::ty::context::TyCtxt>::resolver_for_lowering_raw
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/query/plumbing.rs:406:35
  63:     0x7f2c4f540c55 - <rustc_middle[c1a3e287041cf856]::ty::context::TyCtxt>::resolver_for_lowering
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context.rs:3513:14
  64:     0x7f2c4df77c5d - rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_driver_impl/src/lib.rs:347:25
  65:     0x7f2c4df77c5d - rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt::<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/passes.rs:1020:27
  66:     0x7f2c4df77c5d - <rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>::enter::<rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#1}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context.rs:1651:37
  67:     0x7f2c4df77c5d - rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context::<<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>::enter<rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#1}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  68:     0x7f2c4df77c5d - <std[497fd58be37922e3]::thread::local::LocalKey<core[90b112ef4e25e834]::cell::Cell<*const ()>>>::try_with::<rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context<<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>::enter<rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#1}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/thread/local.rs:513:12
  69:     0x7f2c4df77c5d - <std[497fd58be37922e3]::thread::local::LocalKey<core[90b112ef4e25e834]::cell::Cell<*const ()>>>::with::<rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context<<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>::enter<rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#1}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/thread/local.rs:477:20
  70:     0x7f2c4df77c5d - rustc_middle[c1a3e287041cf856]::ty::context::tls::enter_context::<<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>::enter<rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>::{closure#1}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  71:     0x7f2c4df77c5d - <rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>::enter::<rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context.rs:1651:9
  72:     0x7f2c4df77c5d - <rustc_middle[c1a3e287041cf856]::ty::context::TyCtxt>::create_global_ctxt::<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_middle/src/ty/context.rs:1858:13
  73:     0x7f2c4df77c5d - rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt::<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/passes.rs:987:9
  74:     0x7f2c4df77c5d - <rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[90b112ef4e25e834]::ops::function::FnOnce<(&rustc_session[5163d6f1fc0ec2cd]::session::Session, rustc_middle[c1a3e287041cf856]::ty::context::CurrentGcx, alloc[54afb4ad215f9a5d]::sync::Arc<rustc_data_structures[7f1c5b0d892434a8]::jobserver::Proxy>, &std[497fd58be37922e3]::sync::once_lock::OnceLock<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>, &rustc_data_structures[7f1c5b0d892434a8]::sync::worker_local::WorkerLocal<rustc_middle[c1a3e287041cf856]::arena::Arena>, &rustc_data_structures[7f1c5b0d892434a8]::sync::worker_local::WorkerLocal<rustc_hir[e559e947bda72207]::Arena>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/ops/function.rs:250:5
  75:     0x7f2c4def9763 - <alloc[54afb4ad215f9a5d]::boxed::Box<dyn for<'a> core[90b112ef4e25e834]::ops::function::FnOnce<(&'a rustc_session[5163d6f1fc0ec2cd]::session::Session, rustc_middle[c1a3e287041cf856]::ty::context::CurrentGcx, alloc[54afb4ad215f9a5d]::sync::Arc<rustc_data_structures[7f1c5b0d892434a8]::jobserver::Proxy>, &'a std[497fd58be37922e3]::sync::once_lock::OnceLock<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt<'a>>, &'a rustc_data_structures[7f1c5b0d892434a8]::sync::worker_local::WorkerLocal<rustc_middle[c1a3e287041cf856]::arena::Arena<'a>>, &'a rustc_data_structures[7f1c5b0d892434a8]::sync::worker_local::WorkerLocal<rustc_hir[e559e947bda72207]::Arena<'a>>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}), Output = core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>>> as core[90b112ef4e25e834]::ops::function::FnOnce<(&rustc_session[5163d6f1fc0ec2cd]::session::Session, rustc_middle[c1a3e287041cf856]::ty::context::CurrentGcx, alloc[54afb4ad215f9a5d]::sync::Arc<rustc_data_structures[7f1c5b0d892434a8]::jobserver::Proxy>, &std[497fd58be37922e3]::sync::once_lock::OnceLock<rustc_middle[c1a3e287041cf856]::ty::context::GlobalCtxt>, &rustc_data_structures[7f1c5b0d892434a8]::sync::worker_local::WorkerLocal<rustc_middle[c1a3e287041cf856]::arena::Arena>, &rustc_data_structures[7f1c5b0d892434a8]::sync::worker_local::WorkerLocal<rustc_hir[e559e947bda72207]::Arena>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2})>>::call_once
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/alloc/src/boxed.rs:2206:9
  76:     0x7f2c4def9763 - rustc_interface[240f805485c85136]::passes::create_and_enter_global_ctxt::<core[90b112ef4e25e834]::option::Option<rustc_interface[240f805485c85136]::queries::Linker>, rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}::{closure#2}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/passes.rs:1028:5
  77:     0x7f2c4df6b4e5 - rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_driver_impl/src/lib.rs:340:22
  78:     0x7f2c4df6b4e5 - rustc_interface[240f805485c85136]::interface::run_compiler::<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/interface.rs:543:80
  79:     0x7f2c4df6b4e5 - <core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}::{closure#0}> as core[90b112ef4e25e834]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/panic/unwind_safe.rs:274:9
  80:     0x7f2c4df6b4e5 - std[497fd58be37922e3]::panicking::catch_unwind::do_call::<core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:581:40
  81:     0x7f2c4df6b4e5 - std[497fd58be37922e3]::panicking::catch_unwind::<(), core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:544:19
  82:     0x7f2c4df6b4e5 - std[497fd58be37922e3]::panic::catch_unwind::<core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panic.rs:359:14
  83:     0x7f2c4df6b4e5 - rustc_interface[240f805485c85136]::interface::run_compiler::<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/interface.rs:543:23
  84:     0x7f2c4df52295 - rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals::<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/util.rs:209:17
  85:     0x7f2c4df52295 - rustc_interface[240f805485c85136]::util::run_in_thread_with_globals::<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/util.rs:163:24
  86:     0x7f2c4df52295 - <scoped_tls[ff0a450122c1d024]::ScopedKey<rustc_span[7053f045cc1d2195]::SessionGlobals>>::set::<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>
                               at /rust/deps/scoped-tls-1.0.1/src/lib.rs:137:9
  87:     0x7f2c4df52295 - rustc_span[7053f045cc1d2195]::create_session_globals_then::<(), rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_span/src/lib.rs:143:21
  88:     0x7f2c4df52295 - rustc_interface[240f805485c85136]::util::run_in_thread_with_globals::<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/compiler/rustc_interface/src/util.rs:159:17
  89:     0x7f2c4df52295 - std[497fd58be37922e3]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/sys/backtrace.rs:166:18
  90:     0x7f2c4df7a283 - std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked::<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/thread/lifecycle.rs:91:13
  91:     0x7f2c4df7a283 - <core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}> as core[90b112ef4e25e834]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/panic/unwind_safe.rs:274:9
  92:     0x7f2c4df7a283 - std[497fd58be37922e3]::panicking::catch_unwind::do_call::<core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:581:40
  93:     0x7f2c4df7a283 - std[497fd58be37922e3]::panicking::catch_unwind::<(), core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panicking.rs:544:19
  94:     0x7f2c4df7a283 - std[497fd58be37922e3]::panic::catch_unwind::<core[90b112ef4e25e834]::panic::unwind_safe::AssertUnwindSafe<std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/panic.rs:359:14
  95:     0x7f2c4df7a283 - std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked::<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/thread/lifecycle.rs:89:26
  96:     0x7f2c4df7a283 - <std[497fd58be37922e3]::thread::lifecycle::spawn_unchecked<rustc_interface[240f805485c85136]::util::run_in_thread_with_globals<rustc_interface[240f805485c85136]::util::run_in_thread_pool_with_globals<rustc_interface[240f805485c85136]::interface::run_compiler<(), rustc_driver_impl[2248f57d38587f69]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[90b112ef4e25e834]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /rustc-dev/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/core/src/ops/function.rs:250:5
  97:     0x7f2c5151dfef - <alloc[54afb4ad215f9a5d]::boxed::Box<dyn core[90b112ef4e25e834]::ops::function::FnOnce<(), Output = ()> + core[90b112ef4e25e834]::marker::Send> as core[90b112ef4e25e834]::ops::function::FnOnce<()>>::call_once
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/alloc/src/boxed.rs:2206:9
  98:     0x7f2c5151dfef - <std[497fd58be37922e3]::sys::thread::unix::Thread>::new::thread_start
                               at /rustc/f2c70877a7dbc51cd98e2d5d25209b24d4a586d5/library/std/src/sys/thread/unix.rs:119:17
  99:     0x7f2c4b09698b - <unknown>
 100:     0x7f2c4b11a9cc - <unknown>
 101:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: please attach the file at `/tmp/im/rustc-ice-2025-12-17T17_10_56-561465.txt` to your bug report

query stack during panic:
#0 [resolver_for_lowering_raw] getting the resolver for lowering
end of query stack
error: aborting due to 2 previous errors

Some errors have detailed explanations: E0428, E0433.
For more information about an error, try `rustc --explain E0428`.
```

</p>
</details>

<!--
query stack:

-->

```
note: the unresolved item import `Fish` is defined here
 --> /tmp/crash.rs:2:11
  |
2 |   use A::{::Fish};
  |           ^^^^^^
help: import `Fish` through the re-export
  |
7 |     use A::{{root}}::Fish;
  |         ~~~~~~~~~~~~~~~~~
```

I don't know if it makes sense for `{{root}}` to ever show up in a suggestion tbh.
Does fixing that also fix the ice?


with `nightly-2023-12-05`  we would show
```
error[E0603]: unresolved item import `Fish` is private
 --> /tmp/crash.rs:7:12
  |
7 |     use A::Fish;
  |            ^^^^ private unresolved item import
  |
note: the unresolved item import `Fish` is defined here
 --> /tmp/crash.rs:2:11
  |
2 |   use A::{::Fish};
  |           ^^^^^^
```
but with  `nightly-2023-12-06`, it changed to
```
note: the unresolved item import `Fish` is defined here
 --> /tmp/crash.rs:2:11
  |
2 |   use A::{::Fish};
  |           ^^^^^^
help: import `Fish` through the re-export
  |
7 |     use A::{{root}}::Fish;
  |         ~~~~~~~~~~~~~~~~~
```

could be related to https://github.com/rust-lang/rust/pull/118066 ? cc @estebank

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -2223,7 +2223,11 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
             match binding.kind {
                 NameBindingKind::Import { import, .. } => {
                     for segment in import.module_path.iter().skip(1) {
-                        path.push(segment.ident);
+                        // Don't include `{{root}}` in suggestions - it's an internal symbol
+                        // that should never be shown to users.
+                        if segment.ident.name != kw::PathRoot {
+                            path.push(segment.ident);
+                        }
                     }
                     sugg_paths.push((
                         path.iter().cloned().chain(std::iter::once(ident)).collect::<Vec<_>>(),

--- a/tests/ui/imports/nested-import-root-symbol-150103.rs
+++ b/tests/ui/imports/nested-import-root-symbol-150103.rs
@@ -1,0 +1,13 @@
+// Issue: https://github.com/rust-lang/rust/issues/150103
+// ICE when using `::` at start of nested imports
+// caused by `{{root}}` appearing in diagnostic suggestions
+
+mod A {
+    use Iuse::{ ::Fish }; //~ ERROR failed to resolve: use of unresolved module or unlinked crate
+}
+
+mod B {
+    use A::{::Fish}; //~ ERROR failed to resolve: crate root in paths can only be used in start position
+}
+
+fn main() {}

--- a/tests/ui/imports/nested-import-root-symbol-150103.stderr
+++ b/tests/ui/imports/nested-import-root-symbol-150103.stderr
@@ -1,0 +1,20 @@
+error[E0433]: failed to resolve: use of unresolved module or unlinked crate `Iuse`
+  --> $DIR/nested-import-root-symbol-150103.rs:6:9
+   |
+LL |     use Iuse::{ ::Fish };
+   |         ^^^^ use of unresolved module or unlinked crate `Iuse`
+   |
+help: you might be missing a crate named `Iuse`, add it to your project and import it in your code
+   |
+LL + extern crate Iuse;
+   |
+
+error[E0433]: failed to resolve: crate root in paths can only be used in start position
+  --> $DIR/nested-import-root-symbol-150103.rs:10:13
+   |
+LL |     use A::{::Fish};
+   |             ^ crate root in paths can only be used in start position
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0433`.

